buildtools/composer/composer_test.go

```diff
@@ -1,48 +1,109 @@
 package composer_test
 
 import (
-	"path/filepath"
+	"fmt"
 	"testing"
 
-	"github.com/fossas/fossa-cli/analyzers"
-	"github.com/fossas/fossa-cli/analyzers/php"
-	"github.com/fossas/fossa-cli/module"
-	"github.com/fossas/fossa-cli/pkg"
 	"github.com/stretchr/testify/assert"
+
+	"github.com/fossas/fossa-cli/buildtools/composer"
 )
 
-func TestNoDependenciesNamedPHP(t *testing.T) {
-	// TODO: this really shouldn't require the build tool, but we don't currently
-	// have a non-build-tool method or mock for Composer. We should implement this
-	// by adding resolution from walking `composer.json` dependency manifests.
-	if testing.Short() {
-		t.Skip("Composer requires build tool")
+// The mockRunner type implements composer.Composer.
+type mockRunner struct {
+	showTreeOutput, showOutput, installOutput string
+}
+
+func (mr mockRunner) Show(dir string, args ...string) (stdout string, stderr string, err error) {
+	for _, arg := range args {
+		if arg == "--tree" {
+			return mr.showTreeOutput, "", nil
+		}
 	}
+	return mr.showOutput, "", nil
+}
+
+func (mr mockRunner) Install(dir string, args ...string) (stdout string, stderr string, err error) {
+	return mr.installOutput, "", nil
+}
+
+func TestDependencies(t *testing.T) {
```

Comment:
We have had issues with dependency graphs being constructed incorrectly in the past and I prefer to test the graph a little more explicitly than we have here. This example from the gradle buildtools test is a good way to ensure that every part of the graph is correct. 
https://github.com/fossas/fossa-cli/blob/master/buildtools/gradle/gradle_test.go#L23